sloprovisioning.go

```diff
@@ -93,6 +93,37 @@ func (c *Client) GetVolumesInStorageGroupIterator(ctx context.Context, symID str
 	return c.getVolumeIDsIteratorBase(ctx, symID, query)
 }
 
+// GetVolumeIDsIteratorWithParams returns an iterator of a list of volumes with query parameters
+// For multiple parameters in single field, use ',' to separate the values
+func (c *Client) GetVolumeIDsIteratorWithParams(ctx context.Context, symID string, queryParams map[string]string) (*types.VolumeIterator, error) {
+	defer c.TimeSpent("GetVolumeIDsIteratorWithParams", time.Now())
+	if _, err := c.IsAllowedArray(symID); err != nil {
+		return nil, err
+	}
+
+	query := ""
+	if queryParams != nil {
+		query += "?"
+		for key, val := range queryParams {
+			for _, subVal := range strings.Split(val, ",") {
```

Comment:
in this way, all fields can be added to query, even wrong field; do we need add one supported fields enum  and use switch here ?

Reply:
I was following the practice of snapshots, and invalid field does no harm to the API. Furthermore, since we would sctrictly design the input on the terraform side, there's no problem to leave this as it is.